src/Config.ts

```diff
@@ -57,8 +60,35 @@ export type StreamrClientOptions = Partial<Omit<StrictStreamrClientOptions, 'dat
 const { ControlMessage } = ControlLayer
 const { StreamMessage } = MessageLayer
 
+const validateOverridedEthereumAddresses = (opts: any, propertyPaths: string[]) => {
+    for (const propertyPath of propertyPaths) {
+        if (has(opts, propertyPath)) {
+            const value = get(opts, propertyPath)
+            if (!isAddress(value)) {
+                throw new Error(`${propertyPath} is not a valid Ethereum address`)
+            }
+        }
+    }
+}
+
 /** @internal */
 export default function ClientConfig(opts: StreamrClientOptions = {}) {
+
+    // validate all Ethereum addresses which are required in StrictStreamrClientOptions: if user
+    // overrides a setting, which has a default value, it must be a non-null valid Ethereum address
+    // TODO could also validate
+    // - other optional Ethereum address (if there will be some)
+    // - other overriden options (e.g. regexp check that "restUrl" is a valid url)
+    validateOverridedEthereumAddresses(opts, [
+        'streamrNodeAddress',
+        'tokenAddress',
+        'tokenSidechainAddress',
+        'dataUnion.factoryMainnetAddress',
+        'dataUnion.factorySidechainAddress',
+        'dataUnion.templateMainnetAddress',
+        'dataUnion.templateSidechainAddress'
```

Comment:
Neat, as-is this is an improvement, but note this will silently stop working if a config is renamed or if new EthereumAddress config items are added. Developer always needs to remember to update this list of paths. A shame since we do already have the type information. 

I wonder if we should do something like I described in slack, where we have an unsafe type at API boundaries, then a safe type internally and the only way to get from unsafe to safe type is to pass through the `isAddress` validator.